### PR TITLE
Enhance the dashboard UI with widgets to match the designs

### LINKED ISSUE #44: Enhance the dashboard UI with widgets to match the designs

### PR DESCRIPTION
**Pull request by hiimjin**
# Pull Request Title
Enhance the dashboard UI with widgets to match the designs

## Description
Enhance the dashboard UI with widgets to match the designs

What does this PR fix? You can mention the issue number like
so to reference the issue automatically: "Fixes #44 "

## Screenshot/Recording (if applicable)
Screenshot 2024-10-31 at 12 41 37 PM

### CODE REVIEW

File: Vineyard/Vineyard/Widget.swift [hunk 1]
```diff
@@ -6,121 +6,440 @@
 //
 
 import Foundation
+import SwiftUICore
```
Comment (sankaet8) [inline Vineyard/Vineyard/Widget.swift]: we do not need swiftui core, only swiftUI

File: Vineyard/Vineyard/Widget.swift [hunk 2]
```diff
@@ -6,121 +6,440 @@
...
+    var group: String
+    var isCompleted: Bool = false
 }
 
-struct TodaysTasksWidget: Widget {
-    var todayTasks: [Tsk] = [
-        Tsk(userName: "D", taskText: "Task 1", group: "groupD"),
-        Tsk(userName: "E", taskText: "Task 2", group: "groupE"),
-        Tsk(userName: "F", taskText: "Task 3", group: "groupF")
+//struct TodaysTasksWidgetFull: Widget {
```
Comment (sankaet8) [inline Vineyard/Vineyard/Widget.swift]: Please remove these comments

File: Vineyard/Vineyard/Widget.swift [hunk 3]
```diff
@@ -6,121 +6,440 @@
...
+            .frame(height: 200)
+            .background(Color.purple.opacity(0.35))
+            .cornerRadius(10)
+            .shadow(radius: 5)
         )
     }
 }
 
-struct RecentActivitiesWidget: Widget {
+//struct RecentActivitiesWidgetFull: Widget {
```
Comment (sankaet8) [inline Vineyard/Vineyard/Widget.swift]: Please remove these comments as well

File: Vineyard/Vineyard/Widget.swift
Comment (sankaet8) [inline Vineyard/Vineyard/Widget.swift:1]: Would be nice to split this up into several files each with one type of widget for easier file organization

**Review by sankaet8 — DISMISSED** [review]
please make the changes requested, great work!!

**Review by rahulon12 — CHANGES_REQUESTED** [review]
Can you try using `Grid` instead of multiple `VStack`s and `HStack`s? It has built-in support for col span, etc.

https://developer.apple.com/documentation/swiftui/grid

**Review by rahulon12 — APPROVED** [review]
Looks good for now. We will need to change the widget implementation to make it more flexible for data flow, etc.

**Pull request merged**